### ISSUE TITLE
Feature Request: Add a ban message

### ISSUE THREAD

**Issue by rlmoser99** [TheOdinProject/theodinproject#4666]
### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] The title of this issue follows the `Feature Request: brief description of feature request` format, e.g. `Feature Request: Add a dark mode to the website`
- [ ] Would you like to work on this issue?

### Description of the Feature Request

When an admin needs to ban a user, it would be nice to be able to include a message indicating why their account was banned.

### Acceptance criteria

- [ ] Add an area to write a ban message somewhere in the ban workflow.
- [ ] Send an email with the ban message to the user's provided email address

**Comment by KevinMulhern** [comment]
Thanks for writing this up @rlmoser99. We're got two ban flows atm with both old and new admin operating concurrently. I'll put this on hold until we decomission old admin, that should happen within the next couple of weeks.

Theres two ways going about this,

#### 1. Generic ban email
We don't ban users that often and chances are, they know why they've been banned. We could send a generic ban email with instructions to reach out to the contact email if they'd like find out more / appeal it.

This would be the quickest and easiest way to do it. We'd just need a mailer and hook that into the ban action.

#### 2. Custom message
A custom message is better, but also a lot more work that might not be worth it for the frequency of bans we have.

The UI will be the trickiest part. This is how we choose actions to perform against flags on admin v2:
<img width="678" alt="Screenshot 2024-07-29 at 09 46 54" src="https://github.com/user-attachments/assets/ae3c52c2-c1c4-442a-97c4-c543a36c1aa9">

We'd have to do something like Githubs review UI to include a message with a ban: 
<img width="789" alt="Screenshot 2024-07-29 at 09 47 45" src="https://github.com/user-attachments/assets/03a5c09f-d03a-47b4-b8f4-e613d6d3f51f">

**Comment by rlmoser99** [comment]
Thanks for taking the time to write up the two options! You are right, we do not ban very often, so the first route might be the best option. We could add a list of possible reasons for the ban in the email that we send.

- One of your project submissions contained NSFW or other highly offensive images
- One of your project submissions contained bigotry, such as racism, homophobia, hate speech
- One of your project submissions were plagiarized
- Your account has suspicious votes on your project submissions
- Your account has broken one of our rules (with a link)

**Comment by KevinMulhern** [comment]
It seems difficult to fit ban reasons into the existing UI, but I'm probably not being imaginative enough 😆 

Where in the flag flow do you see ban reasons being slotted in?

**Comment by rlmoser99** [comment]
Sorry for being unclear. I was thinking in the email body of the generic ban email.

> We could send a generic ban email with instructions to reach out to the contact email if they'd like find out more / appeal it.

**Comment by KevinMulhern** [comment]
Sorry, thats my bad, you were clear, I just misread. I think that could work really well, it covers all the bases 💪

**Comment by KevinMulhern** [comment]
Hey @rlmoser99 this is unblocked and ready to work on. Would you like to update the acceptance criteria to match what we talked about with including reasons in the email? 

For most of these stories, I've been writing acceptance criteria [like this](https://github.com/TheOdinProject/theodinproject/pull/4559#issuecomment-2166565535). The exact steps should better help demonstrate the flow we want and reduce any ambiguity.

**Comment by rlmoser99** [comment]
@KevinMulhern When you get a chance can you check out the updated AC? Do you think it would be useful to list Technical considerations, like where the email template files should go or point to another example of the email?

**Comment by KevinMulhern** [comment]
AC looks great to me @rlmoser99 💪 

> Do you think it would be useful to list Technical considerations, like where the email template files should go or point to another example of the email?

Yeah it could be, if you think it will help I'd say go for it. Our user mailer is probably the best example, but the template might not be the best to follow haha.

**Comment by anandhakrishnanaji** [comment]
I'd like to work on this.

**Comment by OmarMoataz** [comment]
Hey guys, can I take this one? @KevinMulhern @rlmoser99

**Comment by KevinMulhern** [comment]
Sorry for the delay! great to see you still around @OmarMoataz, are you still interested in taking this one on? or do you want to pass it on to @ChuqiaoHuang?

**Comment by OmarMoataz** [comment]
@KevinMulhern I would still like to take it.

**Comment by KevinMulhern** [comment]
All yours @OmarMoataz!

**Comment by OmarMoataz** [comment]
Hey, is it theodinprojectcontact@gmail.com or contact@theodinproject.com? @KevinMulhern @rlmoser99

**Comment by OmarMoataz** [comment]
What do you guys think of this?

![Image](https://github.com/user-attachments/assets/80736e23-ff05-4b1f-a80e-2e37b0ba8e46)

**Comment by OmarMoataz** [comment]
Minus the double email 😅

**Comment by KevinMulhern** [comment]
Looking 🔥 @OmarMoataz!

A little bit of feedback:

- Could we move the "For example:" to its own line above the reasons and have it the same font size/style as the reasons.
- Could we bullet point the reasons?

Basically like this:

For example:
- Reason 1
- Reason 2
